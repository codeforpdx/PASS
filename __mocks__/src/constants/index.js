const ENV = {
    VITE_SOLID_IDENTITY_PROVIDER: 'https://solidcommunity.net'
};

// eslint-disable-next-line
export { ENV };