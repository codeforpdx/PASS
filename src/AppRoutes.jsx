// React Imports
import React from 'react';
import { Routes, Route, Navigate, Outlet } from 'react-router-dom';
// Inrupt Imports
import { useSession } from '@hooks';
// Page Imports
import { Home, Clients, Messages, Profile } from './pages';

const ProtectedRoute = ({ isLoggedIn, children }) =>
  isLoggedIn ? children ?? <Outlet /> : <Navigate to="/" replace />;

/**
 * The main application routing for PASS
 *
 * @name AppRoutes
 * @returns {React.JSX.Element} The main routing component for PASS
 */
const AppRoutes = () => {
  const { session } = useSession();
  const restorePath = localStorage.getItem('restorePath');
  const loggedIn = session.info.isLoggedIn;
  const path = loggedIn ? restorePath || '/clients' : '/';

  return (
    <Routes>
      <Route
        exact
        path="/"
        element={session.info.isLoggedIn ? <Navigate to={path} replace /> : <Home />}
      />
      <Route element={<ProtectedRoute isLoggedIn={session.info.isLoggedIn} />}>
        <Route path="/clients" element={<Clients />} />
        <Route path="/messages" element={<Messages />} />
        <Route path="/profile">
          <Route index element={<Profile />} />
          <Route path=":webId" element={<Profile />} />
        </Route>
        <Route path="*" element={<Navigate to={restorePath} replace />} />
      </Route>
    </Routes>
  );
};

export default AppRoutes;
