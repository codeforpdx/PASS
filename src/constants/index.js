import RDF_PREDICATES from './rdf_predicates';
import INTERACTION_TYPES from './interaction_types';

export { RDF_PREDICATES, INTERACTION_TYPES };
