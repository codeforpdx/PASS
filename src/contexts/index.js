import SelectUserContext from './selectUserContext';
import UserListContext, { UserListContextProvider } from './UserListContext';
import MessageContext, { MessageContextProvider } from './MessageContext';

/**
 * The contexts module contains React Contexts that helps with user management
 * and selection
 *
 * @namespace contexts
 */

export {
  SelectUserContext,
  UserListContext,
  MessageContext,
  MessageContextProvider,
  UserListContextProvider
};
export * from './SignedInUserContext';
