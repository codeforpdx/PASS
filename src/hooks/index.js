import useContactsList from './useContactsList';
import useField from './useField';
import useSession from './useSession';
import useStatusNotification from './useStatusNotification';
/**
 * The hooks module contains custom hooks to assist with form handling or status
 * notifications
 *
 * @namespace hooks
 */

export { useContactsList, useField, useStatusNotification, useSession };
