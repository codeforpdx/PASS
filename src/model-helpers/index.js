/**
 * The model-helper module is utility module that helps with the management of
 * users in a users list
 *
 * @memberof utils
 * @name model-helper
 */

export * from './UserList';
export * from './User';
export * from './Document';
export * from './DocumentList';
