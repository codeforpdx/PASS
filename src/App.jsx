// React Imports
import React from 'react';
import { HashRouter as Router } from 'react-router-dom';

// Inrupt Library Imports
import { SessionProvider } from '@inrupt/solid-ui-react';

// Material UI Imports
import { ThemeProvider } from '@mui/material/styles';
import CssBaseline from '@mui/material/CssBaseline';

import theme from './theme';
import UserDataContextProvider from './contexts/UserDataContext';
import Layout from './layouts/Layout';
import AppRoutes from './AppRoutes';

/**
 * @typedef {import("./typedefs").userListObject} userListObject
 */

/**
 * @typedef {import("./typedefs").messageListObject} messageListObject
 */

const App = () => (
  <Router>
    <SessionProvider>
      <CssBaseline />
      <ThemeProvider theme={theme}>
        <Layout ariaLabel="Home Page">
          <UserDataContextProvider>
            <AppRoutes />
          </UserDataContextProvider>
        </Layout>
      </ThemeProvider>
    </SessionProvider>
  </Router>
);

export default App;
