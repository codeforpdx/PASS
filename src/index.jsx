// React Imports
import React from 'react';
import ReactDOM from 'react-dom/client';

// Component Imports
import App from './App';
// Other Imports

const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
