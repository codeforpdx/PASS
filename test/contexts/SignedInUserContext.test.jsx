import { render, cleanup } from '@testing-library/react';
import React, { useContext } from 'react';
import {
  getPodUrlAll,
  getSolidDataset,
  mockSolidDatasetFrom,
  setThing,
  buildThing,
  createThing
} from '@inrupt/solid-client';
import dayjs from 'dayjs';
import { expect, it, afterEach, describe, vi } from 'vitest';
import { SignedInUserContext, SignedInUserContextProvider, SessionContext } from '@contexts';
import { RDF_PREDICATES } from '@constants';
import { fetchProfileInfo } from '../../src/model-helpers';

const TestConsumer = () => {
  const { podUrl } = useContext(SignedInUserContext);

  return <div>{podUrl}</div>;
};

vi.mock('@inrupt/solid-client');

vi.mock('../../src/model-helpers/', async () => {
  const actual = await vi.importActual('../../src/model-helpers/');

  return {
    ...actual,
    fetchProfileInfo: vi.fn()
  };
});

describe('SignedInUserContext', () => {
  afterEach(() => {
    cleanup();
    vi.clearAllMocks();
  });

  it('fetches user data if user is logged in', async () => {
    const newActiveTTL = buildThing(createThing({ name: 'active' }))
      .addDatetime(RDF_PREDICATES.dateModified, dayjs().$d)
      .build();

    const dataset = mockSolidDatasetFrom('https://example.com/pod/PASS/Public/active.ttl');
    getSolidDataset.mockResolvedValue(setThing(dataset, newActiveTTL));

    fetchProfileInfo.mockResolvedValue({ profileInfo: {} });
    getPodUrlAll.mockResolvedValue(['https://example.com/pod/']);
    const { findByText } = render(
      <SessionContext.Provider
        value={{
          session: {
            info: {
              isLoggedIn: true,
              webId: 'https://example.com/pod/profile/card#me'
            }
          }
        }}
      >
        <SignedInUserContextProvider>
          <TestConsumer />
        </SignedInUserContextProvider>
      </SessionContext.Provider>
    );
    const val = await findByText('https://example.com/pod/');
    expect(val).not.toBeNull();
  });
});
