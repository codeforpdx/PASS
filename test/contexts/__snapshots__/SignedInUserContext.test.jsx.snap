// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignedInUserContext > fetches user data if user is logged in 1`] = `
<div>
  <div>
    https://example.com/pod/
  </div>
</div>
`;
